### PR TITLE
Redis track 2 .NET SDK 2022-06-01 API version updates

### PR DESCRIPTION
**Pull request by austintolani**
**Summary**
These changes contain the newly generated code for the `Azure.ResourceManager.Redis` package based on the new 2022-06-01 API version. This generation actually includes new code for changes made in both the 2022-06-01 and 2022-05-01 API versions:

- https://github.com/Azure/azure-rest-api-specs/pull/20869
- https://github.com/Azure/azure-rest-api-specs/pull/19506

**Testing**
I have re-recorded the existing tests using the new API version and added some additional testing coverage.

### CODE REVIEW

File: sdk/redis/Azure.ResourceManager.Redis/CHANGELOG.md [hunk 1]
```diff
@@ -3,12 +3,19 @@
...
-
+- ExportRdbContent has a new parameter PreferredDataArchiveAuthMethod
+- ImportRdbContent hasa new paramter PreferredDataArchiveAuthMethod
+- RedisLinkedServerWithPropertyCreateOrUpdateContent has a new parameter GeoReplicatedPrimaryHostName
+- RedisLinkedServerWithPropertyCreateOrUpdateContent has a new parameter PrimaryHostName
+- RedisLinkedServerWithPropertyData has a new parameter GeoReplicatedPrimaryHostName
+- RedisLinkedServerWithPropertyData has a new parameter PrimaryHostName
 ### Breaking Changes
-
+- Deleting a linked server is now a long running operation
```
Comment (archerzz) [inline sdk/redis/Azure.ResourceManager.Redis/CHANGELOG.md:13]: If there are breaking changes, we need to bump the major version, e.g. `2.0.0-beta.1`. @ArthurMa1978  FYI

File: sdk/redis/Azure.ResourceManager.Redis/src/Generated/RedisLinkedServerWithPropertyData.cs [hunk 2]
```diff
@@ -27,12 +27,16 @@ public RedisLinkedServerWithPropertyData()
...
+        /// <param name="primaryHostName"> The changing DNS name that resolves to the current geo-primary cache among the linked redis caches before or after the Geo Failover. </param>
         /// <param name="provisioningState"> Terminal state of the link between primary and secondary redis cache. </param>
-        internal RedisLinkedServerWithPropertyData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, ResourceIdentifier linkedRedisCacheId, AzureLocation? linkedRedisCacheLocation, RedisLinkedServerRole? serverRole, string provisioningState) : base(id, name, resourceType, systemData)
+        internal RedisLinkedServerWithPropertyData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, ResourceIdentifier linkedRedisCacheId, AzureLocation? linkedRedisCacheLocation, RedisLinkedServerRole? serverRole, string geoReplicatedPrimaryHostName, string primaryHostName, string provisioningState) : base(id, name, resourceType, systemData)
```
Comment (archerzz) [inline sdk/redis/Azure.ResourceManager.Redis/src/Generated/RedisLinkedServerWithPropertyData.cs:33]: There are breaking changes in the interfaces (e.g. parameters are inserted before existing parameters). Here are 2 options:
  - Add manual codes to provide backward compatibility, see compute for reference: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations
- Or you need to bump the major version

**Review by archerzz — CHANGES_REQUESTED** [review]
There are two types of breaking changes in the new version:
1. Two methods are changed to long running
2. New parameters inserted into the constructor of model class

- For item2, you could write customization codes to provide backward compatible constructors. 
- For item1, I think you could also write customization codes to provide backward compatible operation methods. Inside the backward compatible methods, you `await` on the new long running methods.

I'll leave the decision for you. You could write customization codes to provide backward compatibility. Or you could bump the major version.

**Review by archerzz — APPROVED** [review]

**Comment by archerzz** [comment]
> @archerzz I've spoked with my team and we've decided the best approach for us is to bump the major version number. I've updated the version number to 2.0.0 in `CHANGELOG.md`. Is there anything else that I need to do to? Should I be leaving the date as "(Unreleased)" in `CHANGELOG.md`?

- If it's just an update on the codes and you're not going to release the SDK right now, you can leave it with `(Unreleased)`.
- If you're going to release the SDK, you need to update with the release date. 

By the way, the pipeline is broken because the version number in `CHANGELOG.md` is inconsistent with the value in [Azure.ResourceManager.Redis.csproj](https://github.com/Azure/azure-sdk-for-net/blob/71241a89e8e999a62bb14f7b8ac33955533aa0f1/sdk/redis/Azure.ResourceManager.Redis/src/Azure.ResourceManager.Redis.csproj#L3).

**Comment by austintolani** [comment]
Thanks @archerzz, I've updated the CHANGELOG accordingly. My plan is to release the SDK. Can we merge the PR?

**Comment by austintolani** [comment]
@archerzz There is one failing test that seems to be a timeout failure. Are these type of failures common? Is there anything I can do to fix it?

**Comment by archerzz** [comment]
> @archerzz There is one failing test that seems to be a timeout failure. Are these type of failures common? Is there anything I can do to fix it?

No worry. I've restarted the failed job, and it's fine now.

**Comment by austintolani** [comment]
Awesome, thanks for all your help @archerzz

**Pull request merged**